libraries/Microsoft.Bot.Builder.Dialogs.Declarative/Resources/ResourceExplorer.cs

```diff
@@ -116,9 +116,11 @@ public static ResourceExplorer LoadProject(string projectFile, string[] ignoreFo
                 var path = Path.Combine(projectFolder, PathUtils.NormalizePath(node.Attributes["Include"].Value));
                 path = Path.GetFullPath(path);
                 path = Path.GetDirectoryName(path);
-                explorer.AddResourceProvider(new FolderResourceProvider(path, includeSubFolders: true, monitorChanges: monitorChanges));
+                if (Directory.Exists(path))
+                {
+                    explorer.AddResourceProvider(new FolderResourceProvider(path, includeSubFolders: true, monitorChanges: monitorChanges));
+                }
```

Comment:
If the directory does not exist, consider throwing an exception. I think that if a non-existent directory is passed, the best behavior might be to explicitly fail rather than silently succeed but not really register anything.

Reply:
This is checking for installed project reference in a .csproj file. As the runtime is built and installed, the exhaustive search for all project references seems over aggressive, no?

see first line of code pasted below

``` 
  // add project references
            foreach (XmlNode node in xmlDoc.SelectNodes("//ProjectReference"))
            {
                var path = Path.Combine(projectFolder, PathUtils.NormalizePath(node.Attributes["Include"].Value));
                path = Path.GetFullPath(path);
                path = Path.GetDirectoryName(path);
                if (Directory.Exists(path))
                {
                    explorer.AddResourceProvider(new FolderResourceProvider(path, includeSubFolders: true, monitorChanges: monitorChanges));
                }
            }
```